features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/persistence/FlagsStateDeserializer.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.featureflags.internal.persistence
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.flags.featureflags.internal.model.FlagsStateEntry
+import com.datadog.android.flags.featureflags.internal.model.JsonKeys
+import com.datadog.android.flags.featureflags.internal.model.PrecomputedFlag
+import com.datadog.android.flags.featureflags.model.EvaluationContext
+import org.json.JSONException
+import org.json.JSONObject
+
+internal class FlagsStateDeserializer(
+    private val internalLogger: InternalLogger
+) : Deserializer<String, FlagsStateEntry> {
+
+    override fun deserialize(model: String): FlagsStateEntry? {
+        return try {
+            val json = JSONObject(model)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val contextJson = json.getJSONObject(JsonKeys.EVALUATION_CONTEXT.value)
+            val evaluationContext = deserializeEvaluationContext(contextJson)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val flagsJson = json.getJSONObject(JsonKeys.FLAGS.value)
+            val flags = deserializeFlags(flagsJson)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val timestamp = try {
+                json.getLong(JsonKeys.LAST_UPDATE_TIMESTAMP.value)
```

Comment:
maybe `optLong` will be better here? or should we handle missing key here at all?